WordPress/gutenberg
# InputControl: Fix undo when changing padding values

Discovered in #40505 

## What?

This PR solves an issue we have in `InputField` component where it the component receives a new "value" prop, instead of updating its internal state with the new value, it actually does the opposite, it triggers onChange with the old value, causing a potential infinite loop.

## Why?

Undo is broken if you change values in the padding control using "drag". So this is a bug fix.

**Notes**

I've found all the nested chain of components UnitControl -> ValueInput -> NumberControl -> InputField to be too complex. I wonder if all that complexity is worth for a component that is supposed to be simple.

My changes are breaking unit tests. I think it's probably other components (RangeControl and UnitControl) that are misusing InputField but I may be wrong here, I might need some help here @ciampo 

## Testing Instructions

1- Add a group block
2- Click the "padding" input in the sidebar
3- Click and drag the input to change the value
4- While leaving the focus on the input, click "ctrl + z" to undo the changes
5- You'll notice that it undos but very quickly restores to the modified value in trunk
6- in this branch, it undos the value properly

## Screenshots or screencast
